File: typespec/hub/candidacy.go
package hub

import "github.com/psankar/vetchi/typespec/common"

type AddHubCandidacyCommentRequest struct {
	CandidacyID string `json:"candidacy_id" validate:"required"`
	Comment     string `json:"comment"      validate:"required,max=2048"`
}

type MyCandidaciesRequest struct {
	CandidacyStates []common.CandidacyState `json:"candidacy_states" validate:"omitempty,dive,validate_candidacy_state"`
	PaginationKey   *string                 `json:"pagination_key"   validate:"omitempty"`
	Limit           int                     `json:"limit"            validate:"min=0,max=40"`
}

type MyCandidacy struct {
	CandidacyID        string                `json:"candidacy_id"`
	CompanyName        string                `json:"company_name"`
	CompanyDomain      string                `json:"company_domain"`
	OpeningID          string                `json:"opening_id"`
	OpeningTitle       string                `json:"opening_title"`
	OpeningDescription string                `json:"opening_description"`
	CandidacyState     common.CandidacyState `json:"candidacy_state"`
}

File: typespec/hub/candidacy.ts
import { CandidacyState } from '../common/interviews';
import { GetCandidacyCommentsRequest, CandidacyComment } from '../common/candidacies';

export interface AddHubCandidacyCommentRequest {
    candidacy_id: string;
    comment: string;
}

export interface MyCandidaciesRequest {
    candidacy_states?: CandidacyState[];
    pagination_key?: string;
    limit?: number;
}

export interface MyCandidacy {
    candidacy_id: string;
    company_name: string;
    company_domain: string;
    opening_id: string;
    opening_title: string;
    opening_description: string;
    candidacy_state: CandidacyState;
}

export type { GetCandidacyCommentsRequest, CandidacyComment }; 